test/lib/native/testlib_threads.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#ifndef TEST_LIB_NATIVE_THREAD_H
+#define TEST_LIB_NATIVE_THREAD_H
+
+// Header only library for using threads in tests
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <unistd.h>
+#include <pthread.h>
+#endif
+
+extern "C" {
+
+typedef void(*PROCEDURE)(void*);
+
+struct Helper {
+    PROCEDURE proc;
+    void* context;
+};
+
+static void fatal(const char* message) {
+    perror(message);
+    exit(-1);
```

Comment:
Won't work as intended for Windows APIs. I would print the result of `GetLastError()` instead. I would be fine with just omitting the error code too, because I think the old tests did not handle errors either. Or did we catch std::thread exceptions somewhere?

Reply:
The intent was to exit the test with a non-zero exit code, in order to avoid any accidental false positives.

I could return the error code from `GetLastError` and from the respective pthread apis as an exit code instead. Is that what you mean?

Reply:
FWIW, `perror` just prints to `stderr`: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/perror-wperror?view=msvc-170

Reply:
Ah, I see what you mean now. The C standard library function `perror` defines this to also print a textual description of `errno`. https://en.cppreference.com/w/c/io/perror so that won't work for the Windows APIs.

I think the alternative would be to use `FormatMessage` on Windows. I didn't really think that much into this.

Maybe it's clearer to use `fputs` with `stderr` here.